### PR TITLE
T1429

### PR DESCRIPTION
**Pull request by secretrobotron**

### CODE REVIEW

File: cornfield/app.js [hunk 1]
```diff
@@ -117,101 +136,98 @@ function publishRoute( req, res ){
         var projectPath = PUBLISH_DIR + "/" + id + ".html",
             url = PUBLISH_PREFIX + "/" + id + ".html",
             projectData = JSON.parse( project.data ),
-            templateBase = VALID_TEMPLATES[ template ].replace( '{{templateBase}}', TEMPLATES_DIR + '/' );
-
-        fs.readFile( templateBase, 'utf8', function(err, conf){
```
Comment (secretrobotron) [inline cornfield/app.js:122]: NOTE: all of this is just an indentation change.

File: cornfield/app.js [hunk 2]
```diff
@@ -230,7 +246,41 @@ function publishRoute( req, res ){
...
+        project;
+    for ( var i = 0, l = doc.projects.length; i < l; ++i ) {
+      project = doc.projects[ i ];
+      if ( project.template && VALID_TEMPLATES[ project.template ] ) {
+        userProjects.push({
+          _id: project._id,
+          name: project.name,
+          template: project.template,
+          href: templateConfigs[ project.template ].template + 
+            "?savedDataUrl=" + PUBLISH_PREFIX + "/api/project/" + project._id
```
Comment (secretrobotron) [inline cornfield/app.js:274]: Here's the questionable meat.

File: cornfield/app.js [hunk 3]
```diff
@@ -14,6 +14,23 @@ const express = require('express'),
...
       EXPORT_ASSETS = CONFIG.exportAssets;
 
+var templateConfigs = {};
+
+function readTemplateConfig( templateName, templatedPath ) {
+  var configPath = templatedPath.replace( '{{templateBase}}', TEMPLATES_DIR + '/' );
+  fs.readFile( configPath, 'utf8', function( err, conf ) {
+    var configPathBase = configPath.substring( 0, configPath.lastIndexOf( '/' ) );
+    conf = JSON.parse( conf );
+    conf.template = configPathBase + '/' + conf.template;
```
Comment (secretrobotron) [inline cornfield/app.js:24]: Attempted to use npm module `url` here, but was a bit wonky. File new ticket for follow up if desired, please.

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @jbuck received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/10a2d1cfa21276e/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @jbuck received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/13d90380ae627ab/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/10a2d1cfa21276e/output.txt

Total script time: 0.48 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/13d90380ae627ab/output.txt

Total script time: 0.45 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @jbuck received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/3c1cb84f47ba998/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @jbuck received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/05b59b0451aa221/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/3c1cb84f47ba998/output.txt

Total script time: 0.33 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/05b59b0451aa221/output.txt

Total script time: 0.76 mins

Server is running on http://pr558.butyr.org/templates/test.html

File: cornfield/app.js (hunk 2 above)
Comment (jbuck) [inline cornfield/app.js:274]: It looks pretty good actually

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @humphd received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/b7b3f78e6e8cffd/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/b7b3f78e6e8cffd/output.txt

Total script time: 1.20 mins

Server is running on http://pr558.butyr.org/templates/test.html

**Pull request merged**